src/ipc/handler.c

```diff
@@ -815,6 +817,147 @@ static int ipc_glb_gdb_debug(uint32_t header)
 
 }
 
+#if CONFIG_PROBE
+static inline int ipc_probe_init(uint32_t header)
+{
+	struct sof_ipc_probe_dma_add_params *params = _ipc->comp_data;
+	int dma_set_up = (params->hdr.size - sizeof(*params)) /
+			 sizeof(struct probe_dma);
+
+	tracev_ipc("ipc_probe_init()");
+
+	return probe_init(dma_set_up ? params->probe_dma : NULL);
+}
+
+static inline int ipc_probe_deinit(uint32_t header)
+{
+	tracev_ipc("ipc_probe_deinit()");
+
+	return probe_deinit();
+}
+
+static inline int ipc_probe_dma_add(uint32_t header)
+{
+	struct sof_ipc_probe_dma_add_params *params = _ipc->comp_data;
+	uint32_t dmas_count = (params->hdr.size - sizeof(*params)) /
+			      sizeof(struct probe_dma);
+
+	tracev_ipc("ipc_probe_dma_add()");
+
+	return probe_dma_add(dmas_count, params->probe_dma);
+}
+
+static inline int ipc_probe_dma_remove(uint32_t header)
+{
+	struct sof_ipc_probe_dma_remove_params *params = _ipc->comp_data;
+	uint32_t tags_count = (params->hdr.size - sizeof(*params)) /
+			      sizeof(uint32_t);
+
+	tracev_ipc("ipc_probe_dma_remove()");
+
+	return probe_dma_remove(tags_count, params->stream_tag);
+}
+
+static inline int ipc_probe_point_add(uint32_t header)
+{
+	struct sof_ipc_probe_point_add_params *params = _ipc->comp_data;
+	uint32_t probes_count = (params->hdr.size - sizeof(*params)) /
+				sizeof(struct probe_point);
+
+	tracev_ipc("ipc_probe_point_add()");
+
+	return probe_point_add(probes_count, params->probe_point);
+}
+
+static inline int ipc_probe_point_remove(uint32_t header)
+{
+	struct sof_ipc_probe_point_remove_params *params = _ipc->comp_data;
+	uint32_t probes_count = (params->hdr.size - sizeof(*params)) /
+				sizeof(uint32_t);
+
+	tracev_ipc("ipc_probe_point_remove()");
+
+	return probe_point_remove(probes_count, params->buffer_id);
+}
+
+static int ipc_probe_info(uint32_t header)
+{
+	uint32_t cmd = iCS(header);
+	struct sof_ipc_probe_info_params *params = _ipc->comp_data;
+	int ret;
+
+	tracev_ipc("ipc_probe_get_data()");
+
+	switch (cmd) {
+	case SOF_IPC_PROBE_DMA_INFO:
+		ret = probe_dma_info(params, SOF_IPC_MSG_MAX_SIZE);
+		break;
+	case SOF_IPC_PROBE_POINT_INFO:
+		ret = probe_point_info(params, SOF_IPC_MSG_MAX_SIZE);
+		break;
+	default:
+		trace_ipc_error("ipc_probe_info() error: Invalid probe INFO command = %u",
+				cmd);
+		ret = -EINVAL;
+	}
+
+	if (ret < 0) {
+		trace_ipc_error("ipc_probe_info() error: cmd %u failed", cmd);
+		return ret;
+	}
+
+	/* write data to the outbox */
+	if (params->rhdr.hdr.size <= MAILBOX_HOSTBOX_SIZE &&
+	    params->rhdr.hdr.size <= SOF_IPC_MSG_MAX_SIZE) {
+		mailbox_hostbox_write(0, params, params->rhdr.hdr.size);
+		ret = 1;
```

Comment:
usually success means a return value of zero, why is this 1 here?

Reply:
Positive value (here 1), at this stage, means custom reply was given, and handler doesn't post generic responses.
Please refer to `ipc_cmd()` in handler.c:1292.

Reply:
well then describe what you do... this sort of code can be understood as a bug when it's a feature.

Reply:
I've just used existing API which was designed exactly this way, for this purpose. I think it might be documented somewhere around/within that `ipc_cmd()` function, but I don't think that this should be included in this PR.